### DISCUSSION TITLE
HashRouter possible?

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [nandorojo/solito#418] [Q&A]
Is it possible to use a HashRouter?

**Comment by nandorojo** [comment]
hey, what is that?

**Comment by NilsBaumgartner1994** [comment]
If Routing is not done by „…com/app/settings/„ but instead with „…con/#app/settings/„This allows deploying to GitHub pages with routing and reopening pages.Beste Grüße,Nils BaumgartnerAm 31.07.2023 um 19:50 schrieb Fernando Rojo ***@***.***>:﻿
hey, what is that?

**Reply by nandorojo** [reply]
sorry i don’t understand

**Comment by NilsBaumgartner1994** [comment]
No problem.Typically Routing in apps is path based right?Like example.com/orders will show you all ordersOr example.com/order/5 will show you the order with id 5This is great since users and developers know for each resource where to find it.When you want to deploy your app via GitHub pages you will face some issues.Your application will be published under username.GitHub.io/RepoNameBut GitHub won’t support normal Routing for example:Nandorojo.GitHub.Io/solito/ordersWould result in an 404 error since GitHub would search for a file at this endpoint. Instead you have to use „Hash based Routing“.Therefore the route would be nandorojo.GitHub.io/solito/#/ordersSince # are not considered as path splits but parameters the router would virtually split by this. Which allows you to be able to publish your application via GitHub pages.A better explanation ***@***.***/deploying-a-create-react-app-with-routing-to-github-pages-f386b6ce84c2Beste Grüße,Nils BaumgartnerAm 31.07.2023 um 21:51 schrieb Fernando Rojo ***@***.***>:﻿
sorry i don’t understand

**Comment by derek-primumco** [comment]
I have no idea. I know Gatsby is commonly used with GitHub Pages. https://www.google.com/search?q=gatsby+github+pages

I don't believe Solito is officially supported for any Next.js deployment other than Vercel, so you might have to hack it up yourself if you absolutely need to deploy the web side of Solito with another provider like Netlify or GitHub Pages.

**Reply by nandorojo** [reply]
yeah just use vercel

**Comment by NilsBaumgartner1994** [comment]
Yeah, that’s just not a good answer.So solito can‘t HashRoute … yet?Hash based Routing is a nice thing. Please consider it implementing.Beste Grüße,Nils BaumgartnerAm 31.07.2023 um 22:47 schrieb Fernando Rojo ***@***.***>:﻿
yeah just use vercel

**Comment by derek-primumco** [comment]
I would also advise anyone not using Vercel for Next.js to switch to Vercel, so that is a good answer. :blush:

But on the topic of HashRouting, it does sound nice. Not to speak for the repo maintainer (Fernando), but I think you'd be welcome to open a PR or add additional findings to this discussion thread for the community to work on.

My understanding is that keeping updated with Expo SDK updates is the main goal for Solito at this time, so something like adding hash routing just isn't on the roadmap.

**Reply by nandorojo** [reply]
i don’t actually know what it is, but if it works with next it should work with solito. presumably it would need simple rewrites or redirects with a pound sign from next.js
